### ISSUE TITLE
About loc_pred's meaning?

### ISSUE THREAD

**Issue by randf102** [open-mmlab/mmdetection#749]
Hi, I need to do some post-processing to the anchor location prediction(Just in the testing mode I will change it, training mode is like the original version ). So I tried to modify the forward_single function in [guided_anchor_head.py](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/anchor_heads/guided_anchor_head.py#L199).
I use the weight I have trained before. But after I modify the code, the detecting result did not change.  I also simply try to replace all the location_prediction to zero. 
Like this
```
def forward_single(self, x):
        loc_pred = self.conv_loc(x)
        ###
        a = torch.zeros([1,1,loc_pred.shape[2],loc_pred.shape[3]], dtype=torch.float64, device=device)
        loc_pred = a
        ###

        shape_pred = self.conv_shape(x)
        x = self.feature_adaption(x, shape_pred)
```

And it still got the same output to the original result. 

Am I mistake the meaning of "**loc_pred**", or **loc_pred** is not the only one to decide where is the object's center?

Thank you!

**Comment by myownskyW7** [comment]
If you use 'python setup.py develop' to setup mmdetection? If you use 'python setup.py install', you local change will not affect the code actually runs.

**Comment by myownskyW7** [comment]
In Line 337: loc_pred = loc_pred.sigmoid().detach(), we apply a sigmoid function on loc_pred. If loc_pred is zero before sigmoid, after sigmoid it will be 0.5.

**Comment by randf102** [comment]
Thank you for your reply!
But if all the loc_pred is set to zero, all the loc_pred should get the same value after sigmoid. Why does the output result seem still has the capability to detect objects?

**Comment by myownskyW7** [comment]
It is because when loc_pred is set to zero, loc_pred.sigmoid() will be 0.5. Then all locations will be kept since they have scores >= loc_filter_thr.
You can try to print loc_pred to check if it is actually set to 0. If you setup mmdetection with 'python setup.py install' rather than 'python setup.py develop', your local changes will not affect the code which actually runs.

**Comment by randf102** [comment]
Thank you for your kind assistance! I have successfully modified the codes and got the expected results. 
By the way, there is a very small typo in [getting started.md](https://github.com/open-mmlab/mmdetection/blob/master/GETTING_STARTED.md#high-level-apis-for-testing-images)

**High-level APIs for testing images.**
Here is an example of building the model and test given images.
```
from mmdet.apis import init_detector, inference_detector, show_result

config_file = 'configs/faster_rcnn_r50_fpn_1x.py'
checkpoint_file = 'checkpoints/faster_rcnn_r50_fpn_1x_20181010-3d1b3351.pth'

# build the model from a config file and a checkpoint file
model = init_detector(config_file, checkpoint_file, device='cuda:0'))
                                                                    ^
                                                                   HERE
```

thank you!
